src/main/java/com/PBA/budgetservice/facade/AccountFacadeImpl.java

```diff
@@ -0,0 +1,38 @@
+package com.PBA.budgetservice.facade;
+
+import com.PBA.budgetservice.controller.request.AccountCreateRequest;
+import com.PBA.budgetservice.mapper.AccountMapper;
+import com.PBA.budgetservice.persistance.model.Account;
+import com.PBA.budgetservice.persistance.model.dtos.AccountDto;
+import com.PBA.budgetservice.service.AccountService;
+import org.springframework.stereotype.Component;
+
+import java.util.UUID;
+
+@Component
+public class AccountFacadeImpl implements AccountFacade {
+    private final AccountService accountService;
+    private final AccountMapper accountMapper;
+
+    public AccountFacadeImpl(AccountService accountService, AccountMapper accountMapper) {
+        this.accountService = accountService;
+        this.accountMapper = accountMapper;
+    }
+
+    @Override
+    public AccountDto createAccount(AccountCreateRequest accountCreateRequest) {
+        this.validateAccountAlreadyExists(accountCreateRequest);
+        Account accountToCreate = accountMapper.toAccount(accountCreateRequest);
+
+        Account savedAccount = accountService.addAccount(accountToCreate);
+        return accountMapper.toAccountDto(savedAccount);
+    }
+
+    private void validateAccountAlreadyExists(AccountCreateRequest accountCreateRequest) {
+        UUID userUid = accountCreateRequest.getUserUid();
+        String currency = accountCreateRequest.getCurrency();
+        if (accountService.accountExists(userUid, currency)) {
+            //todo: throw AccountAlreadyExists exception and handle it
```

Comment:
and who is gonna do it ? :)))

Reply:
I figured that it had nothing to do with the given task for creating an account (happy flow), so I didn't do it yet. Should I commit the implementation for this in the current PR, or make a new PR for the rest exception handling task and push it there?

Reply:
You can throw the exception for now and the handling will be done in the task for rest exception handling.